src/GameSelector/UserGamesList.tsx

```diff
@@ -1,33 +1,74 @@
-import {
-  Heading,
-  Box,
-  HStack,
-  Flex,
-  CheckIcon,
-  SmallCloseIcon,
-  Stack,
-  View,
-  Text,
-  Button,
-} from 'native-base';
-import React from 'react';
-import { Board } from '../GameState/Board';
+import { Box, View, Text, FlatList, Flex, Badge } from 'native-base';
+import React, { useState } from 'react';
+import { Pressable } from 'react-native';
 import { Game } from '../interface';
+import { UserGame } from './UserGame';
 
 interface UserGamesListProps {
-  navigateToGameState: (id: number) => void;
+  navigateToGameState: (id: number | undefined) => void;
   userGames: Game[] | undefined;
 }
 
 export const UserGamesList = ({
   userGames,
   navigateToGameState,
 }: UserGamesListProps) => {
+  const [filter, setFilter] = useState<'all' | 'inProgress' | 'finished'>(
+    'inProgress',
+  );
+
+  const filterGames = (filter: string) => {
```

Comment:
The name filter is not explicit. gameState is more appropriate because we understand directly that the filter will be based on the game state.